[phantomjs] Initial plan and tests

MindNumbing opened the pull request:
Signed-off-by: Steven Marshall <SMarshall@chef.io>

Commit:
[phantomjs] Initial plan and tests

Signed-off-by: Steven Marshall <SMarshall@chef.io>

skpaterson commented on controls/pcre_works.rb:
```diff
@@ -0,0 +1,36 @@
+
+  phantomjs_works = command("/bin/phantomjs -v")
+  describe phantomjs_works do
+    its('stdout') { should match /[0-9]+.[0-9]+.[0-9]+/ }
```
Small suggestion, to match on the exact version string could use:
```
    its('stdout') { should match /#{target_dir.split('/')[5]}/ }
```

skpaterson reviewed (approved):
Thanks @MindNumbing - one small suggestion but fine to proceed as is.

Commit:
[phantomjs] Various refactors to existing code

Added inspec.lock to .gitignore
Updated README.md to latest binary format
Added plan_name to attributes.yml
Updates controls to add more prescriptive version checking.
Updated inspec.yml to latest version

Signed-off-by: Steven Marshall <SMarshall@chef.io>
